common/logger/src/lib.rs

```diff
@@ -1,73 +1,90 @@
-        }
-    }
-}
+pub use serde_json::json;
 
-#[cfg(test)]
-mod tests {
-    use super::{init, Flag};
-    use log::{debug, error, info, trace, warn};
+pub fn init(
+    filter: String,
+    log_to_console: bool,
+    console_show_file_and_line: bool,
+    log_to_file: bool,
+    metrics: bool,
+    log_path: PathBuf,
+) {
+    let console = ConsoleAppender::builder()
+        .encoder(Box::new(PatternEncoder::new(
+            if console_show_file_and_line {
+                "[{d} {h({l})} {t} {f}:{L}] {m}{n}"
+            } else {
+                "[{d} {h({l})} {t}] {m}{n}"
+            },
+        )))
+        .build();
 
-    #[test]
-    fn test_debug() {
-        init(Flag::Test);
-        debug!("Hello World!");
-        debug!("Hello {}", "World!");
-        debug!("Hell{} {}", "o", "World!");
-        debug!("Hell{} World", 0);
-    }
+    let file = FileAppender::builder()
+        .encoder(Box::new(JsonEncoder::new()))
+        .build(log_path.join("muta.log"))
+        .unwrap();
 
-    #[test]
-    fn test_error() {
-        init(Flag::Test);
-        error!("Hello World!");
-        error!("Hello {}", "World!");
-        error!("Hell{} {}", "o", "World!");
-        error!("Hell{} World", 0);
-    }
+    let metrics_appender = FileAppender::builder()
+        .encoder(Box::new(JsonEncoder::new()))
+        .build(log_path.join("metrics.log"))
+        .unwrap();
 
-    #[test]
-    fn test_info() {
-        init(Flag::Test);
-        info!("Hello World!");
-        info!("Hello {}", "World!");
-        info!("Hell{} {}", "o", "World!");
-        info!("Hell{} World", 0);
+    let mut root_builder = Root::builder();
+    if log_to_console {
+        root_builder = root_builder.appender("console");
     }
-
-    #[test]
-    fn test_warn() {
-        init(Flag::Test);
-        warn!("Hello World!");
-        warn!("Hello {}", "World!");
-        warn!("Hell{} {}", "o", "World!");
-        warn!("Hell{} World", 0);
+    if log_to_file {
+        root_builder = root_builder.appender("file");
     }
+    let level_filter = match filter.as_ref() {
+        "off" => LevelFilter::Off,
+        "error" => LevelFilter::Error,
+        "info" => LevelFilter::Info,
+        "warn" => LevelFilter::Warn,
+        "debug" => LevelFilter::Debug,
+        "trace" => LevelFilter::Trace,
+        f => {
+            println!("invalid logger.filter {}, use info", f);
+            LevelFilter::Info
+        }
+    };
+    let root = root_builder.build(level_filter);
 
-    #[test]
-    fn test_trace() {
-        init(Flag::Test);
-        trace!("Hello World!");
-        trace!("Hello {}", "World!");
-        trace!("Hell{} {}", "o", "World!");
-        trace!("Hell{} World", 0);
-    }
+    let metrics_logger = Logger::builder().additive(false).appender("metrics").build(
+        "metrics",
+        if metrics {
+            LevelFilter::Trace
+        } else {
+            LevelFilter::Off
```

Comment:
Does this mean that `trace` information is lost?

The PR may be missing some necessary instructions.

Reply:
> Does this mean that trace logs will be discarded?

No, only the `metrics` target is discarded.

> The PR may be missing some necessary instructions.

That makes sense. I'll add a readme file.